src/udev/udev-manager.c

```diff
@@ -871,19 +872,16 @@ static int on_ctrl_msg(UdevCtrl *uctrl, UdevCtrlMessageType type, const UdevCtrl
                 log_debug("Received udev control message (RELOAD)");
                 manager_reload(manager, /* force = */ true);
                 break;
-        case UDEV_CTRL_SET_ENV: {
-                const char *eq;
-
-                eq = strchr(value->buf, '=');
-                if (!eq) {
-                        log_error("Invalid key format '%s'", value->buf);
-                        return 1;
+        case UDEV_CTRL_SET_ENV:
+                /* The restriction for udev property is not clear. Let's apply the one for environment variable here. */
+                if (!env_assignment_is_valid(value->buf)) {
```

Comment:
i think doing it like this, but maybe introduce a helper for this:

```c
/* udev properties are conceptually close to environment variables, let's validate assignments the same way */
#define udev_property_assignment_is_valid(x) env_assignment_is_valid(x)
```

in some header (or here?)